modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -1920,15 +1922,26 @@ public ObjectEntry updateStatus(
 			ServiceContext serviceContext)
 		throws PortalException {
 
-		if (objectEntry.getStatus() == status) {
+		if ((objectEntry.getDisplayDate() == null) &&
+			(objectEntry.getStatus() == status)) {
+
 			return objectEntry;
 		}
 
 		ObjectEntry originalObjectEntry = (ObjectEntry)objectEntry.clone();
 
 		Date date = new Date();
+		Date displayDate = objectEntry.getDisplayDate();
 		Date expirationDate = objectEntry.getExpirationDate();
 
+		if ((status == WorkflowConstants.STATUS_APPROVED) &&
```

Comment:
double check this condition according to the AC, please

Reply:
This is also for AC3 and to AC8.